### PR TITLE
Fix tests under -Ydelambdafy:method

### PR DESCRIPTION
**Pull request by lrytz**
We don't currently run a nightly build using `-Ydelambdafy:method`, that's why some tests diverged.

For this PR, a successful `-Ydelambdafy:method` build is [here](https://scala-webapps.epfl.ch/jenkins/view/All/job/scala-checkin-manual/1354/console)

After this PR is merged, I will add the `Ydelambdafy:method` flag to the [nightly genbcode build](https://scala-webapps.epfl.ch/jenkins/view/2.11.x/job/scala-nightly-genbcode-2.11.x/)

### CODE REVIEW

**Comment by gkossakowski** [comment]
Would enabling `-Ydelambdafy:method` by default in 2.12.x branch be enough to get test coverage?

**Comment by lrytz** [comment]
@gkossakowski I would still like to enable it in the `-Ybackend:GenBCode` build on 2.11.x, since development happens on this branch.

The IDE failures can be ignored, they are due to some upgrades in their build script (confirmed by @dragos).

Review by @gkossakowski :)

File: test/files/run/repl-javap-outdir-funs/run-repl_7.scala [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import scala.tools.partest.JavapTest
 
 object Test extends JavapTest {
+  // note the '-fun': it makes :javap search for some anonfun.
+  // for that reason, this test has a flags file that forces delambdafy:inline (doesn't allow :method)
```
Comment (som-snytt) [inline test/files/run/repl-javap-outdir-funs/run-repl_7.scala:5]: override transformSettings to do it inline.  I did that this morning for the fix to javap command (not pushed yet).

File: test/files/run/t6260c.check [hunk 2]
```diff
@@ -1,5 +1,9 @@
 f(C@2e)
 
+#partest !-Ydelambdafy:method
 Test$$anonfun$$apply
+#partest -Ydelambdafy:method
+Test$lambda$1$$apply
+#partest
```
Comment (som-snytt) [inline test/files/run/t6260c.check:7]: That flag-flipping thing works out OK.

**Comment by lrytz** [comment]
ping review @gkossakowski

**Comment by retronym** [comment]
LGTM

**Pull request merged**